hiss/application/models.py

```diff
@@ -286,3 +288,48 @@ def clean(self):
             raise exceptions.ValidationError("First name can't contain any numbers")
         if any(char.isdigit() for char in self.last_name):
             raise exceptions.ValidationError("Last name can't contain any numbers")
+
+
+class WaveManager(models.Manager):
+    def next_wave(self, dt: datetime.datetime = timezone.now()):
+        """
+        Returns the next INACTIVE wave, if one exists. For the CURRENT active wave, use
+        `active_wave`.
+        """
+        qs = self.get_queryset().filter(start__gt=dt).order_by("start")
+        return qs.first()
+
+    def active_wave(self, dt: datetime.datetime = timezone.now()):
+        """
+        Returns the CURRENTLY active wave, if one exists. For the next INACTIVE wave, use
+        `next_wave`.
+        """
+        qs = self.get_queryset().filter(start__lte=dt, end__gt=dt).order_by("start")
+        return qs.first()
+
+
+class Wave(models.Model):
+    """
+    Representation of a registration period. `Application`s must be created during
+    a `Wave`, and are automatically associated with a wave through the `Application`'s `pre_save` handler.
+    """
+
+    start = models.DateTimeField()
+    end = models.DateTimeField()
+    num_days_to_rsvp = models.IntegerField()
+
+    objects = WaveManager()
+
+    def clean(self):
+        super().clean()
+        if self.start >= self.end:
+            raise exceptions.ValidationError(
+                {"start": "Start date can't be after end date."}
+            )
+        for wave in Wave.objects.all():
```

Comment:
We actually don't want to fetch every single `Wave` here, because that includes _this_ one. We want to do something like:

```suggestion
        for wave in Wave.objects.exclude(pk=self.pk).all():
```

[Docs for `exclude`](https://docs.djangoproject.com/en/2.2/ref/models/querysets/#exclude)